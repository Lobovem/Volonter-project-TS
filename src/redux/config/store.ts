import { configureStore } from '@reduxjs/toolkit';
import contentSlice from '../slice/contentSlice';

export const store = configureStore({
  reducer: {
    content: contentSlice,
  },
});

// Выведение типов `RootState` и `AppDispatch` из хранилища
export type RootState = ReturnType<typeof store.getState>;
// Выведенные типы: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
