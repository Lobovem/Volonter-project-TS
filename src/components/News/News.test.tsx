import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../redux/config/store';
import { News } from './News';
import { MemoryRouter } from 'react-router-dom';

describe('News', () => {
  test('renders News component', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <News />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
