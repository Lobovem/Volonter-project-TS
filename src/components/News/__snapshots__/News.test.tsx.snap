// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News renders News component 1`] = `
<div>
  <div
    class="sectionNews"
  >
    <h3
      class="sectionNews__decor"
    >
      Новини
    </h3>
    <div
      class="sectionNews__inner"
    >
      <div
        class="sectionNews__textWrap"
      >
        <h2
          class="titleMain"
        >
          Наша мета - вирішення нагальних потреб Сил оборони України
        </h2>
        <p
          class="sectionNews__desc"
        >
          UkrHumBat надає пріоритет в допомозі військовослужбовцям та членам їх сімей, а також, за можливості, допомагає цивільним особам, які опинились у зв'язку із війною в скрутному становищі.
        </p>
        <a
          class="sectionNews__link"
          href="/news"
        >
          Читати всі статті
        </a>
      </div>
    </div>
  </div>
</div>
`;
